### ISSUE TITLE
bug: 🐝 Cannot create target folder since 4.7.1 on self-hosted runner (`rsync: --mkpath: unknown option`)

### ISSUE THREAD

**Issue by marcus-bcl** [JamesIves/github-pages-deploy-action#1753]
### Describe the bug

Since upgrading to 4.7.1, the deploy action now fails with `rsync: --mkpath: unknown option`.

The `--mkpath` option was added in rsync 3.2.3, but our self-hosted runner (based on https://github.com/myoung34/docker-github-actions-runner) only has 3.1.3.  Do we need to upgrade rsync for this to work or would it be possible for the action to support older versions?

### Reproduction Steps

Use a self-hosted runner with rsync <3.2.3.

### Logs

```shell
/usr/bin/rsync -q -av --checksum --progress --mkpath ... --delete --exclude CNAME --exclude .nojekyll --exclude .ssh --exclude .git --exclude .github
rsync: --mkpath: unknown option
rsync error: syntax or usage error (code 1) at main.c(1601) [client=3.1.3]
```

https://github.com/ministryofjustice/hmpps-probation-integration-e2e-tests/actions/runs/12115120629/job/33776527330#step:12:68

### Workflow

```yml
- name: Publish HTML report
  uses: JamesIves/github-pages-deploy-action@dc18a3c6b46d56484cb63f291becd7ed4f0269b9 # v4
  with:
    folder: folder
    target-folder: target-folder
```

https://github.com/ministryofjustice/hmpps-probation-integration-e2e-tests/actions/runs/12115120629/workflow#L176-L181

**Comment by JamesIves** [comment]
What OS version is your self hosted runner using?

**Comment by marcus-bcl** [comment]
> What OS version is your self hosted runner using?

Ubuntu 20.04

**Comment by Borda** [comment]
we may have similar problem over here:
```
╭━━━╮╭╮╭╮╱╭╮╱╱╭╮╱╱╭━━━╮
    ┃╭━╮┣╯╰┫┃╱┃┃╱╱┃┃╱╱┃╭━╮┃
    ┃┃╱╰╋╮╭┫╰━╯┣╮╭┫╰━╮┃╰━╯┣━━┳━━┳━━┳━━╮
    ┃┃╭━╋┫┃┃╭━╮┃┃┃┃╭╮┃┃╭━━┫╭╮┃╭╮┃┃━┫━━┫
    ┃╰┻━┃┃╰┫┃╱┃┃╰╯┃╰╯┃┃┃╱╱┃╭╮┃╰╯┃┃━╋━━┃
    ╰━━━┻┻━┻╯╱╰┻━━┻━━╯╰╯╱╱╰╯╰┻━╮┣━━┻━━╯
    ╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╭━╯┃
    ╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╱╰━━╯
    ╭━━━╮╱╱╱╱╱╭╮╱╱╱╱╱╱╱╱╭━━━╮╱╱╭╮
    ╰╮╭╮┃╱╱╱╱╱┃┃╱╱╱╱╱╱╱╱┃╭━╮┃╱╭╯╰╮
    ╱┃┃┃┣━━┳━━┫┃╭━━┳╮╱╭╮┃┃╱┃┣━┻╮╭╋┳━━┳━╮
    ╱┃┃┃┃┃━┫╭╮┃┃┃╭╮┃┃╱┃┃┃╰━╯┃╭━┫┃┣┫╭╮┃╭╮╮
    ╭╯╰╯┃┃━┫╰╯┃╰┫╰╯┃╰━╯┃┃╭━╮┃╰━┫╰┫┃╰╯┃┃┃┃
    ╰━━━┻━━┫╭━┻━┻━━┻━╮╭╯╰╯╱╰┻━━┻━┻┻━━┻╯╰╯
    ╱╱╱╱╱╱╱┃┃╱╱╱╱╱╱╭━╯┃
    ╱╱╱╱╱╱╱╰╯╱╱╱╱╱╱╰━━╯
    

    💖 Support: https://github.com/sponsors/JamesIves
    📣 Maintained by James Ives: https://jamesiv.es

    🚀 Getting Started Guide: https://github.com/JamesIves/github-pages-deploy-action
    ❓ Discussions / Q&A: https://github.com/JamesIves/github-pages-deploy-action/discussions
    🔧 Report a Bug: https://github.com/JamesIves/github-pages-deploy-action/issues
Checking configuration and starting deployment… 🚦
Deploying using Deploy Token… 🔑
Configuring git…
/usr/bin/git config --global --add safe.directory /home/runner/work/utilities/utilities
/usr/bin/git config user.name Borda
/usr/bin/git config user.email 6035[28](https://github.com/Lightning-AI/utilities/actions/runs/12135515450/job/33835493833#step:8:29)4+Borda@users.noreply.github.com
/usr/bin/git config core.ignorecase false
/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
Unable to unset previous git config authentication as it may not exist, continuing…
/usr/bin/git remote rm origin
/usr/bin/git remote add origin ***github.com/Lightning-AI/utilities.git
Git configured… 🔧
Starting to commit changes…
/usr/bin/git ls-remote --heads ***github.com/Lightning-AI/utilities.git refs/heads/gh-pages
c838d37a7c4d040508ff59fc549c69ce99e2dbbf	refs/heads/gh-pages
Creating worktree…
/usr/bin/git fetch --no-recurse-submodules --depth=1 origin gh-pages
From https://github.com/Lightning-AI/utilities
 * branch            gh-pages   -> FETCH_HEAD
 * [new branch]      gh-pages   -> origin/gh-pages
/usr/bin/git worktree add --no-checkout --detach github-pages-deploy-action-temp-deployment-folder
Preparing worktree (detached HEAD 716e0b4)
/usr/bin/git checkout --orphan gh-pages origin/gh-pages
Previous HEAD position was 716e0b4 build(deps): bump JamesIves/github-pages-deploy-action from 4.6.9 to 4.7.1 (#338)
Switched to a new branch 'gh-pages'
/usr/bin/git config --global --add safe.directory /home/runner/work/utilities/utilities/github-pages-deploy-action-temp-deployment-folder
/usr/bin/chmod -R +rw /home/runner/work/utilities/utilities/docs/build/html
Creating target folder if it doesn't already exist… 📌
/usr/bin/rsync -q -av --checksum --progress --mkpath /home/runner/work/utilities/utilities/docs/build/html/. github-pages-deploy-action-temp-deployment-folder/docs --delete --exclude CNAME --exclude .ssh --exclude .git --exclude .github
rsync: --mkpath: unknown option
rsync error: syntax or usage error (code 1) at main.c(1601) [client=3.1.3]
Running post deployment cleanup jobs… 🗑️
/usr/bin/git checkout -B github-pages-deploy-action/r[29](https://github.com/Lightning-AI/utilities/actions/runs/12135515450/job/33835493833#step:8:30)tii4yy
Switched to a new branch 'github-pages-deploy-action/r29tii4yy'
/usr/bin/chmod -R +rw github-pages-deploy-action-temp-deployment-folder
/usr/bin/git worktree remove github-pages-deploy-action-temp-deployment-folder --force
Error: The deploy step encountered an error: The process '/usr/bin/rsync' failed with exit code 1 ❌
```
and the job: https://github.com/Lightning-AI/utilities/actions/runs/12135515450/job/33835493833#step:8:80

---

UPDATE: seems bumping OS to `ubuntu-latest` helped

**Comment by JamesIves** [comment]
I've got a potential fix for this in the works that I'll work to get up by EOD

**Comment by JamesIves** [comment]
Should be fixed in [4.7.2](https://github.com/JamesIves/github-pages-deploy-action/releases/tag/v4.7.2)
